src/content/docs/en/recipes/making-toolbar-apps.mdx

```diff
@@ -0,0 +1,229 @@
+
+In order to use relative paths for entrypoints, you should always use the `import.meta.url` variable to get the path to the current file and resolve the path to the entrypoint from there to make sure it works regardless of where the integration is used.
+:::
+
+You can now use this integration in your project by adding it to the `integrations` array in your `astro.config.mjs` file.
+
+```js title="astro.config.mjs"
+import { defineConfig } from 'astro/config';
+import myIntegration from './my-integration.ts';
+
+export default defineConfig({
+  integrations: [myIntegration],
+})
+```
+
+## Creating the app
+
+Dev toolbar apps are defined using the `defineToolbarApp` function from the `astro/toolbar` module. This function takes an object with an `init` function that will be called when the dev toolbar app is loaded. 
+
+```ts title="app.ts"
+import { defineToolbarApp } from "astro/toolbar";
+
+export default defineToolbarApp({
+    init() {
+      // ...
+    },
+});
+```
+
+The `init()` function receives a standard [ShadowRoot](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot) as its first argument, which you can use to create a UI for your dev toolbar app. Elements can be created and added to the ShadowRoot using the standard DOM API.
+
+```ts title="app.ts" ins={5-7}
+import { defineToolbarApp } from "astro/toolbar";
+
+export default defineToolbarApp({
+    init(canvas) {
+      const h1 = document.createElement('h1');
+      h1.textContent = 'Hello World!';
+      canvas.appendChild(h1);
+    },
+});
+```
+
+In most cases, the `init()` function will contain the vast majority of your dev toolbar app's logic, as this function will create the UI for your app, set up event listeners, etc. In order to make it easier to do common operations, `init()` also includes two additional arguments: `app` and `server`.
+
+```ts title="app.ts" ins=/(app),/ ins=/(server)\\)/
+import { defineToolbarApp } from "astro/toolbar";
+
+export default defineToolbarApp({
+    init(canvas, app, server) {
+      // Create a new text node every time the app is toggled on or off
+      app.onAppToggled(({ state }) => {
+        const text = document.createTextNode(
+          `The app is now ${state ? "enabled" : "disabled"}!`,
+        );
+        canvas.appendChild(text);
+      });
+      
+      // Send a message to the server
+      server.send("my-app:hello-server", {message: "Hello, server!"});
+    },
+});
+```
+
+These arguments provide multiple utilities and helpers to interact with the dev toolbar and the server, such as executing code when the app is toggled on or off, showing a notification dot on the app, sending messages to the server, and more.
+
+For more information on the hooks and helpers available in the `init()` function, see the [Astro Toolbar API documentation](https://docs.astro.build/api/astro/toolbar).
```

Comment:
```suggestion
For more information on the hooks and helpers available in the `init()` function, see the [Astro Toolbar API documentation](/en/reference/dev-toolbar-app-reference/).
```
Now you're just making things up!  :smile:

Reply:
I put it as a placeholder while writing and forgot to change it, ha